packages/apps/api/src/handlers/user.ts

```diff
@@ -791,63 +611,61 @@ export async function deleteUser(
   request: Request,
   response: Response
 ): Promise<void> {
-  const userId: number = getTargetUserId(request);
+  const userId = getTargetUserId(request);
 
   const auth0Delete: APIResponse<{}> = await auth0Client.deleteUser(userId);
   if (auth0Delete.status !== ResponseStatus.Success) {
-    if (auth0Delete.status === ResponseStatus.NotFound) {
-      response.status(404).json(toMessage(auth0Delete.message));
-    } else {
-      response.status(500).json(toMessage(auth0Delete.message));
-    }
-    return;
+    throw clientResponseToHttpError(auth0Delete);
   }
 
   const patronDelete: APIResponse<{}> = await sierraClient.deletePatronRecord(
     userId
   );
   if (patronDelete.status !== ResponseStatus.Success) {
-    if (patronDelete.status === ResponseStatus.NotFound) {
-      response.status(404).json(toMessage(patronDelete.message));
-    } else {
-      response.status(500).json(toMessage(patronDelete.message));
-    }
-    return;
+    throw clientResponseToHttpError(patronDelete);
   }
 
   response.sendStatus(204);
 }
 
 function extractSourceIp(request: Request): string {
   if (!request.apiGateway) {
-    throw new Error(
-      "API Gateway event and context data doesn't exist on request"
-    );
+    throw new HttpError({
+      status: 404,
+      message: "API Gateway event and context data doesn't exist on request",
+    });
   }
   return request.apiGateway?.event.requestContext.identity.sourceIp;
 }
 
 function getTargetUserId(request: Request): number {
   if (request.params.user_id === 'me') {
     if (userIsAdmin(request)) {
-      throw new Error('Administrator users cannot operate on themselves');
+      throw new HttpError({
+        status: 404,
+        message: 'Administrator users cannot operate on themselves',
```

Comment:
Would a 403 Forbidden be more appropriate than a 404 here?

Reply:
Hmm yeah probably